web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -96,14 +97,47 @@
             studyIds, sampleIds, attributes.stream().filter(a -> a.getClinicalDataType().equals(ClinicalDataType.SAMPLE))
             .map(a -> a.getAttributeId()).collect(Collectors.toList()), ClinicalDataType.SAMPLE);
-        List<ClinicalDataCountItem> resultForPatientAttributes = clinicalDataService.fetchClinicalDataCounts(studyIds, sampleIds, 
+        List<ClinicalDataCountItem> resultForPatientAttributes = clinicalDataService.fetchClinicalDataCounts(studyIds, sampleIds,
         attributes.stream().filter(a -> a.getClinicalDataType().equals(ClinicalDataType.PATIENT))
             .map(a -> a.getAttributeId()).collect(Collectors.toList()), ClinicalDataType.PATIENT);
         combinedResult.addAll(resultForSampleAttributes);
         combinedResult.addAll(resultForPatientAttributes);
         return new ResponseEntity<>(combinedResult, HttpStatus.OK);
     }
 
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
+    @RequestMapping(value = "/oncokb/fetch", method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch annotation data by study view filter")
+    public ResponseEntity<List<ClinicalDataCountItem>> fetchAnnotationData(
+        @ApiParam(required = true, value = "Annotation data filter")
+        @Valid @RequestBody(required = false)  OncoKBDataCountFilter oncoKBDataCountFilter,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
+        @Valid @RequestAttribute(required = false, value = "interceptedOncoKBDataCountFilter") OncoKBDataCountFilter interceptedOncoKBDataCountFilter) {
+
+        List<OncoKBDataFilter> attributes = interceptedOncoKBDataCountFilter.getAttributes();
+        StudyViewFilter studyViewFilter = interceptedOncoKBDataCountFilter.getStudyViewFilter();
+        if (attributes.size() == 1) {
+            studyViewFilterUtil.removeSelfFromFilter(attributes.get(0).getAttributeId(), studyViewFilter);
+        }
+        List<SampleIdentifier> filteredSampleIdentifiers = studyViewFilterApplier.apply(studyViewFilter);
+
+        List<OncoKBDataCountItem> combinedResult = new ArrayList<>();
+        if (filteredSampleIdentifiers.isEmpty()) {
+            return new ResponseEntity<>(combinedResult, HttpStatus.OK);
+        }
+        List<String> studyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
+        List<ClinicalDataCountItem> resultForSampleAttributes = clinicalDataService.fetchClinicalDataCounts(
```

Comment:
Here is basically need to get mutation data first then return OncoKBDataCountItem based on mutation data.

Reply:
You should start to use `getMutationsInMultipleMolecularProfilesByAnnotation` in mutationService